src/ua_securechannel.h

```diff
@@ -29,36 +31,62 @@ struct ChunkEntry {
     UA_ByteString bytes;
 };
 
-/* For chunked responses */
-typedef struct {
-    UA_SecureChannel *channel;
-    UA_UInt32 requestId;
-    UA_UInt32 messageType;
-    UA_UInt16 chunksSoFar;
-    size_t messageSizeSoFar;
-    UA_Boolean final;
-    UA_StatusCode errorCode;
-} UA_ChunkInfo;
-
 struct UA_SecureChannel {
+    UA_Boolean temporary; // this flag is set to false if the channel is fully opened.
     UA_MessageSecurityMode  securityMode;
     UA_ChannelSecurityToken securityToken; // the channelId is contained in the securityToken
     UA_ChannelSecurityToken nextSecurityToken; // the channelId is contained in the securityToken
-    UA_AsymmetricAlgorithmSecurityHeader clientAsymAlgSettings;
-    UA_AsymmetricAlgorithmSecurityHeader serverAsymAlgSettings;
+    UA_AsymmetricAlgorithmSecurityHeader remoteAsymAlgSettings;
+    UA_AsymmetricAlgorithmSecurityHeader localAsymAlgSettings;
+
+    /** The active security policy and context of the channel */
+    const UA_SecurityPolicy *securityPolicy;
+    void *securityContext;
+
+    /** Stores all available security policies that may be used when establishing a connection. */
+    UA_SecurityPolicies availableSecurityPolicies;
```

Comment:
We are in a temporary channel when this is needed.
In UA_SecureChannel_processAsymmetricOPNChunk we need to check which policy we are using. This is done by iterating through the available policies and matching the policyURI.
If we wanted to pass it here, we'd have to pass it through a lot of functions, so i don't think this is an option. I'll check tomorrow if there's a better alternative. Otherwise this will have to do i guess.